openff/toolkit/topology/molecule.py

```diff
@@ -4739,6 +4853,17 @@ def remap(self, mapping_dict, current_to_new=True):
         # move any properties across
         new_molecule._properties = deepcopy(self._properties)
 
+        # remap the atom map
+        try:
+            new_molecule.properties["atom_map"] = {
+                cur_to_new.get(k, k): v
+                for k, v in new_molecule.properties["atom_map"].items()
+            }
+        except (KeyError, AttributeError):
+            # If the atom map is missing or is not a dictionary, don't worry
+            # about it
+            pass
```

Comment:
Could you make this more explicit with "ifs" for the specific cases we're guarding for? This seems logically right but takes me a while to understand (even with the helpful comment).